core/modules/filters/mediaquery.js

```diff
@@ -0,0 +1,97 @@
+/*\
+title: $:/core/modules/filters/mediaquery.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator for evaluating CSS media queries
+
+\*/
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+// Store active media query listeners
+var mediaQueryListeners = {};
+var listenerCounter = 0;
+
+/*
+Export our filter function
+*/
+exports.mediaquery = function(source,operator,options) {
+	var results = [],
+		mediaQuery = operator.operand,
+		widget = options.widget;
+	
+	// Only evaluate in browser environment
+	if($tw.browser && mediaQuery && widget) {
+		try {
```

Comment:
There is the if() statement, which should catch all problems. We need to make sure, that the variables are initialised. So why do we still need a try/catch statement here.

Reply:
As far as I can see. `window.matchMedia()` does not throw any exceptions

Reply:
I'll remove the try catch

Reply:
Wait, the try catch is indeed needed. Invalid media queries throw exceptions.

Reply:
Let me test, I'm not quite sure

Reply:
You're right @pmario , it doesn't throw exceptions. I've removed the try/catch statement